packages/gatsby-plugin-feed/src/createFeed.js

```diff
@@ -0,0 +1,64 @@
+import url from 'url'
+import { defaultsDeep, get } from 'lodash'
+import RSS from 'rss'
+
+const defaultOptions = {
+  generator: `Gatsby`,
+  title: `RSS Feed`,
+}
+
+export default function createFeed({
+  query: {
+    site: {
+      siteMetadata,
+    },
+    entries,
+    ...rest
+  },
+  output,
+}) {
+  // Ensure that siteMetadata is not undefined
+  const metadata = siteMetadata || {}
+  const options = { ...defaultOptions }
+
+  // Assign the default feed options. Note that we can skip existence
+  // checks here because node-rss ignores undefined key values.
+  options.author = metadata.author
+  options.title = metadata.title
+  options.description = metadata.description
+
+  // Derive site_url and feed_url from the Gatsby site's siteUrl value
+  if (metadata.siteUrl) {
+    options.site_url = metadata.siteUrl
+    options.feed_url = url.resolve(metadata.siteUrl, output)
+  }
+
+  // @TODO: generate custom namespaces and elements for media objects
+  // specified in YAML frontmatter.
+  if (hasMedia(entries)) {
+    options.custom_namespaces = {
+      itunes: `https://www.itunes.com/dtds/podcast-1.0.dtd`,
+    }
+
+    options.custom_elements = {
+
+    }
+  }
+
+  // @TODO: is this useful? We should think of an example that would use this.
+  const context = defaultsDeep(options, rest)
```

Comment:
The original idea behind the `context` object was that it contained options (global ones merged with locally provided `feed`-specific overrides) which were then passed to both the `setup` and `serialize` functions.

`setup` would then return all necessary configuration for the `RSS` object. This essentially allowed the user to configure the `RSS` object as they pleased by only implementing a single function if the other defaults were fine.

`serialize` was much the same story. It received context and returned an array of `items` that could then be appended to the feed object returned by `RSS`.

I think the API I originally proposed was a bit much for generating your basic feed of articles, but it wasn't for other things. I'm afraid removing it entirely goes a bit further than I'd like to go. The plugin's current defaults aren't sensible, but I think a mix of simple defaults and easy overrides is the best policy.